core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -89,14 +104,29 @@ oppia.factory('explorationData', [
           // changes applied. This makes a force-refresh necessary when changes
           // are discarded, otherwise the exploration-with-draft-changes
           // (which is cached here) will be reused.
+          console.log(errorCallback);
           return (
             EditableExplorationBackendApiService.fetchApplyDraftExploration(
             explorationId).then(function(response) {
               $log.info('Retrieved exploration data.');
               $log.info(response);
               draftChangeListId = response.draft_change_list_id;
               explorationData.data = response;
-
+              var draftChanges = LocalStorageService.getExplorationDraft(
```

Comment:
I've had a go at that but with slightly different methods from what you've suggested. What do you think? I wasn't sure about the toJSON method on it's prototype as I'm not sure when it will be used (as the domain object will only be converted to JSON when saving to localStorage which is the same time as when the domain object is created so we'd have something like ExplorationDraftObjectFactory.createFromJSON().toJSON()). Is that what we want or is the way it is currently implemented okay? Thanks!